kuma/api/v1/views.py

```diff
@@ -133,6 +134,19 @@ def document_api_data(doc=None, ensure_contributors=False, redirect_url=None):
     }
 
 
+# Get all waffle flags, switches and samples in advance. Each call to
+# the whoami API will query the current value of each of them. This code
+# for querying the models and then querying the setting for each
+# model is copied from waffle.views._generate_waffle_js.
+#
+# Note that if we upgrade django-waffle, version 15 introduces a pluggable
+# flag model, and the approved way to query all flags will then become:
+#    all_waffle_flags = get_waffle_flag_model().get_all()
+all_waffle_flags = Flag.get_all()
+all_waffle_switches = Switch.get_all()
+all_waffle_samples = Sample.get_all()
```

Comment:
This here makes me nervous. First because it feels bad form to have a module global mutable sitting around simply be importing the module. 
Ideally, we shouldn't be touching the MySQL until Django decides to boot things up. 

Secondly, having data loaded into the module namespace makes it clunky to write tests that change flags, switches, and samples on a test-by-test basis.

Reply:
What do you think about using a memoizing function or something instead? Like django's lru_cache function.

Reply:
Yes, I can change this to call get_all() the first time these are needed, especially if doing it this way is causing a test failure.